lib/Translation_unit.ml

```diff
@@ -208,30 +209,29 @@ let print_error ?(fmt = Format.err_formatter) conf ~input_name error =
                        --no-parse-docstrings.\n\
                        %!"
                       Location.print_loc loc (ellipsis_cmt s))
-          | `Comment_dropped l when not conf.Conf.quiet ->
+          | `Comment_dropped l when not quiet ->
               List.iter l ~f:(fun Cmt.{txt= msg; loc} ->
                   Format.fprintf fmt
                     "%!@{<loc>%a@}:@,\
                      @{<error>Error@}: Comment (* %s *) dropped.\n\
                      %!"
                     Location.print_loc loc (ellipsis_cmt msg))
           | `Cannot_parse ((Syntaxerr.Error _ | Lexer.Error _) as exn) ->
-              if Conf.debug then Location.report_exception fmt exn
+              if debug then Location.report_exception fmt exn
           | `Warning50 l ->
-              if Conf.debug then
+              if debug then
                 List.iter l ~f:(fun (l, w) -> Compat.print_warning l w)
           | _ -> () ) ;
-          if Conf.debug then
+          if debug then
             List.iter l ~f:(fun (msg, sexp) ->
                 Format.fprintf fmt "  %s: %s\n%!" msg (Sexp.to_string sexp))
       | exn ->
           Format.fprintf fmt
             "  BUG: unhandled exception. Use [--debug] for details.\n%!" ;
-          if Conf.debug then Format.fprintf fmt "%s\n%!" (Exn.to_string exn)
-      )
+          if debug then Format.fprintf fmt "%s\n%!" (Exn.to_string exn) )
 
-let check_all_locations fmt cmts_t =
-  if Conf.debug then
+let check_all_locations ~debug fmt cmts_t =
```

Comment:
given that the function is `if <argument> then ...`, I think it would make sense to move the branch to the call site

Reply:
Good idea. That makes the code slightly easier to read.